gnomad/sample_qc/pipeline.py

```diff
@@ -262,36 +345,58 @@ def annotate_sex(
+            - observed_homs (int64): Expected number of homozygotes. Calculated using hl.impute_sex.
+
+        If `infer_karyotype`:
+            - X_karyotype (str): Sample's chromosome X karyotype.
+            - Y_karyotype (str): Sample's chromosome Y karyotype.
+            - sex_karyotype (str): Sample's sex karyotype.
+
+    .. note::
+
+            In order to infer sex karyotype (`infer_karyotype`=True), one of `compute_fstat` or
+            `use_gaussian_mixture_model` must be set to True.
+
+    :param mtds: Input MatrixTable or VariantDataset.
+    :param is_sparse: Whether input MatrixTable is in sparse data format.
     :param excluded_intervals: Optional table of intervals to exclude from the computation.
     :param included_intervals: Optional table of intervals to use in the computation. REQUIRED for exomes.
-    :param normalization_contig: Which chromosome to use to normalize sex chromosome coverage. Used in determining sex chromosome ploidies.
-    :param sites_ht: Optional Table to use. If present, filters input MatrixTable to sites in this Table prior to imputing sex,
-                    and pulls alternate allele frequency from this Table.
+    :param normalization_contig: Which chromosome to use to normalize sex chromosome coverage. Used in determining sex
+        chromosome ploidies.
+    :param sites_ht: Optional Table to use. If present, filters input MatrixTable to sites in this Table prior to
+        imputing sex, and pulls alternate allele frequency from this Table.
     :param aaf_expr: Optional. Name of field in input MatrixTable with alternate allele frequency.
-    :param gt_expr: Name of entry field storing the genotype. Default: 'GT'
-    :param f_stat_cutoff: f-stat to roughly divide 'XX' from 'XY' samples. Assumes XX samples are below cutoff and XY are above cutoff.
-    :param float aaf_threshold: Minimum alternate allele frequency to be used in f-stat calculations.
+    :param gt_expr: Name of entry field storing the genotype. Default is 'GT'.
+    :param f_stat_cutoff: f-stat to roughly divide 'XX' from 'XY' samples. Assumes XX samples are below cutoff and XY
+        are above cutoff.
+    :param aaf_threshold: Minimum alternate allele frequency to be used in f-stat calculations.
     :param variants_only_x_ploidy: Whether to use depth of only variant data for the x ploidy estimation.
     :param variants_only_y_ploidy: Whether to use depth of only variant data for the y ploidy estimation.
+    :param compute_fstat: Whether to compute f-stat. Default is True.
+    :param infer_karyotype: Whether to infer sex karyotypes. Default is True.
+    :param use_gaussian_mixture_model: Whether to use gaussian mixture model to split samples into 'XX' and 'XY'
+        instead of f-stat.
     :return: Table of samples and their imputed sex karyotypes.
     """
     logger.info("Imputing sex chromosome ploidies...")
 
+    if infer_karyotype and not (compute_fstat or use_gaussian_mixture_model):
```

Comment:
As is no, the default is to use f-stat, but that can only be done if you first compute the fstat. If `use_gaussian_mixture_model` is set it will use the gaussian mixture model, but you can still compute f-stat even if it's not used for the karyotype inference.